crates/proof-of-sql/src/sql/ast/multiply_expr_test.rs

```diff
@@ -0,0 +1,344 @@
+        smallint("b", [2_i16, 0]),
+    ]);
+    let t = "sxt.t".parse().unwrap();
+    let accessor = OwnedTableTestAccessor::<InnerProductProof>::new_from_table(t, data, 0, ());
+    let ast: ProofPlan<RistrettoPoint> = dense_filter(
+        vec![(
+            multiply(column(t, "a", &accessor), column(t, "b", &accessor)),
+            "c".parse().unwrap(),
+        )],
+        tab(t),
+        equal(column(t, "b", &accessor), const_bigint(2)),
+    );
+    let verifiable_res: VerifiableQueryResult<InnerProductProof> =
+        VerifiableQueryResult::new(&ast, &accessor, &());
+    assert!(matches!(
+        verifiable_res.verify(&ast, &accessor, &()),
+        Err(QueryError::Overflow)
+    ));
+}
+
+// select a * b as c from sxt.t where b == 0
+#[test]
+fn overflow_in_nonselected_rows_doesnt_error_out() {
+    let data = owned_table([
+        smallint("a", [i16::MAX, i16::MIN + 1]),
+        smallint("b", [2_i16, 0]),
+    ]);
+    let t = "sxt.t".parse().unwrap();
+    let accessor = OwnedTableTestAccessor::<InnerProductProof>::new_from_table(t, data, 0, ());
+    let ast: ProofPlan<RistrettoPoint> = dense_filter(
+        vec![(
+            multiply(column(t, "a", &accessor), column(t, "b", &accessor)),
+            "c".parse().unwrap(),
+        )],
+        tab(t),
+        equal(column(t, "b", &accessor), const_bigint(0)),
+    );
+    let verifiable_res: VerifiableQueryResult<InnerProductProof> =
+        VerifiableQueryResult::new(&ast, &accessor, &());
+    exercise_verification(&verifiable_res, &ast, &accessor, t);
+    let res = verifiable_res.verify(&ast, &accessor, &()).unwrap().table;
+    let expected_res = owned_table([smallint("c", [0_i16])]);
+    assert_eq!(res, expected_res);
+}
+
+// select a, b from sxt.t where a * b >= 0
+#[test]
+fn overflow_in_where_clause_doesnt_error_out() {
+    let data = owned_table([
+        bigint("a", [i64::MAX, i64::MIN + 1]),
+        smallint("b", [2_i16, 1]),
+    ]);
+    let t = "sxt.t".parse().unwrap();
+    let accessor = OwnedTableTestAccessor::<InnerProductProof>::new_from_table(t, data, 0, ());
+    let ast: ProofPlan<RistrettoPoint> = dense_filter(
+        cols_expr_plan(t, &["a", "b"], &accessor),
+        tab(t),
+        gte(
+            multiply(column(t, "a", &accessor), column(t, "b", &accessor)),
+            const_bigint(0),
+        ),
+    );
+    let verifiable_res: VerifiableQueryResult<InnerProductProof> =
+        VerifiableQueryResult::new(&ast, &accessor, &());
+    exercise_verification(&verifiable_res, &ast, &accessor, t);
+    let res = verifiable_res.verify(&ast, &accessor, &()).unwrap().table;
+    let expected_res = owned_table([bigint("a", [i64::MAX]), smallint("b", [2_i16])]);
+    assert_eq!(res, expected_res);
+}
+
+// select a * b as c from sxt.t
+#[test]
+fn result_expr_can_overflow_more() {
+    let data = owned_table([
+        bigint("a", [i64::MAX, i64::MIN, i64::MAX, i64::MIN]),
+        bigint("b", [i64::MAX, i64::MAX, i64::MIN, i64::MIN]),
```

Comment:
I'll make sure to have tests for boundary conditions like this for new features as well